Add course link and "..." dropdown to nav bar

julieg18 opened the pull request:
* refactors nav popup code
* adds `...` dropdown to nav bar (holds items that don't fit on the nav bar)
* add "Course" nav item

Fixes #2955

julieg18 commented on src/components/LayoutHeader/Nav/LinkItems/index.tsx:
```diff
@@ -45,158 +55,91 @@ const navLinkItemsData: Array<INavLinkData | INavLinkPopupData> = [
+  (item as INavLinkPopupData).popupName !== undefined
+
+const LinkItems: React.FC = () => {
+  const communityPopup = usePopup()
```
I went ahead and created a `usePopup` hook that handles popup functionality. The popup code in the nav was already overly complex and adding a third popup would have made things even more complicated 😅

julieg18 commented on src/components/LayoutHeader/Nav/LinkItems/styles.module.css:
```diff
@@ -66,6 +78,18 @@
     }
   }
 
+  &.other {
```
Here's what the popup looks like: 

![image](https://user-images.githubusercontent.com/43496356/153026740-0c6d7cf4-c6ea-4b35-9c1e-78d652e4c4d1.png)


Is it good enough for a first iteration? cc @yalozhkin

julieg18 commented on src/components/LayoutHeader/Nav/LinkItems/index.tsx:
```diff
@@ -1,33 +1,43 @@
 }
 
-const navLinkItemsData: Array<INavLinkData | INavLinkPopupData> = [
+export const navLinkItemsData: Array<INavLinkData | INavLinkPopupData> = [
```
I wasn't sure where "Learn" should go in the nav. For now, I placed it after "Support" :)

rogermparent reviewed (approved):
Looks fine by me, especially as a first iteration before we try unifying all the navs.

Are we okay with "Community" losing its dropdown when in this menu? I don't see a problem with it, but others may think differently. That said, if we want to change it we can do so in a follow up, since this is still a step up from before where it wasn't present at all!

shcheklein commented:
Thanks, @julieg18 ! Can we move the Learn link to the Docs., Blog? (unless you discussed this other folks and this decision was made based of that). It's not a strong opinion, but it makes sense to me to keep them nearby.

julieg18 commented:
> Thanks, @julieg18 ! Can we move the Learn link to the Docs., Blog? (unless you discussed this other folks and this decision was made based of that). It's not a strong opinion, but it makes sense to me to keep them nearby.

If I'm understanding your question correctly, you want to move it to after Blog? I moved it there but let me know if I misunderstood :)

> We already have Learn under the community dropdown, so we should consider changing either of these.

You're right! Plus, as we mentioned before, Learn is kind of an unclear name anyway. I renamed it to "Course" for now! 

> I think we also need to find a proper place on the hamburger menu as well.

Thanks for catching that! I'll update it.

> Q: should we consider adding a new window SVG icon similar to what we have with items on other tools menus?

I'm not sure I understand your question 🤔 Where exactly do you want to add a new window SVG icon?

yathomasi commented:
> I'm not sure I understand your question 🤔 Where exactly do you want to add a new window SVG icon?

![screenshot-dvc-org-add-course-conn-nsrmv2 herokuapp com-2022 02 09-14_03_38](https://user-images.githubusercontent.com/20840228/153154345-f619fa76-98f6-489f-a843-1455e5d744bf.png)

I meant something like this like we are using on `Other tools`. But, I am not sure we would want that.

julieg18 commented:
> I meant something like this like we are using on Other tools. But, I am not sure we would want that.

Oh, ok! Thanks for clarifying further! Personally, I don't think it's needed. But @iterative/websites, what do you think?

shcheklein commented:
Yeah, good point @yathomasi , but I also don't think we need it in this specific case. I would consider adding the link only if it's a completely external website, in this case it's kinda internal link (ecosystem of the websites), so it's fine to avoid it here. And probably in general, menu is to tight on space that it makes sense to sacrifice some consistency here.
